### PR TITLE
Add toggle for post sticky status on editor #1313

### PR DESCRIPTION
**Pull request by jasonagnew**
Added a toggle under Post Settings -> Status & Visibility for post sticky status.

http://share.agnew.co/lqC7JP

Should this option show if the post is not set to Public?
If the post is set to Private or Password Protected should we set Sticky back to false?

Fixes: #1313

### CODE REVIEW

File: editor/sidebar/post-sticky/index.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+/**
+ * Internal dependencies
+ */
+import { getEditedPostAttribute } from '../../selectors';
+import { editPost } from '../../actions';
+
+function PostSticky( { postSticky = false, instanceId, ...props } ) {
+	const stickyToggleId = 'post-sticky-toggle-' + instanceId;
+
+	const onTogglePostSticky = () => props.editPost( { sticky: ! postSticky } );
```
Comment (youknowriad) [inline editor/sidebar/post-sticky/index.js]: Minor: I'd personally bake this in the `mapStateToDispatch` providing a `onTogglePostSticky` prop instead that way the `PostSticky` presentational component would deal only with the sticky value and a simple callback. Doesn't need to know about `editPost`.

**Review by youknowriad — CHANGES_REQUESTED** [review]
Thanks for the PR. This works great.

One point though needs to be addressed. I think the `sticky` toggle is only enabled for the "post" post type, we should make sure to hide the toggle for other post types

**Comment by jasonagnew** [comment]
@youknowriad Thanks for the feedback, I've made those changes. I wasn't completely sure best way to do `onTogglePostSticky` in `mapStateToDispatch` but I gave it a go.

File: editor/sidebar/post-sticky/index.js [hunk 2]
```diff
@@ -36,8 +38,15 @@ function PostSticky( { postSticky = false, instanceId, ...props } ) {
 export default connect(
 	( state ) => {
 		return {
+			postType: getEditedPostAttribute( state, 'type' ),
```
Comment (youknowriad) [inline editor/sidebar/post-sticky/index.js]: Minor: Should we use the newly introduced `getCurrentPostType` selector?

File: editor/sidebar/post-sticky/index.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+	( state ) => {
+		return {
+			postType: getEditedPostAttribute( state, 'type' ),
+			postSticky: getEditedPostAttribute( state, 'sticky' ),
+		};
+	},
+	( dispatch ) => {
+		return {
+			onTogglePostSticky( postSticky ) {
+				dispatch( editPost( { sticky: ! postSticky } ) );
```
Comment (youknowriad) [inline editor/sidebar/post-sticky/index.js]: Yeah, you're right, this looks a bit weird to have  `!` here. Thinking we should rename `onTogglePostSticky` to `onUpdateSticky` and directly pass the updated value. Thoughts?

**Comment by jasonagnew** [comment]
Hey @youknowriad Thanks for your patience. I updated it to use the new `getCurrentPostType` selector and I think calling it `onUpdateSticky` and moving the logic means it works the same as `PostSchedule`

**Review by youknowriad — APPROVED** [review]
Nice 👍 Let's get this merged.

Thanks for working on this.

**Pull request merged**